### ISSUE TITLE
`assert_type` errors reported when type checking is off

### ISSUE THREAD

**Issue by DetachHead** [microsoft/pylance-release#2982]
## Environment data

-   Language Server version: v2022.6.30
-   OS and version: windows 10 21h2
-   Python version (& distribution if applicable, e.g. Anaconda): 3.10 normal

## Code Snippet

```python
from typing_extensions import assert_type

assert_type(1, str)
```

## Repro Steps

1. disable python type checking in settings
    ![image](https://user-images.githubusercontent.com/57028336/176589203-0a1d7323-0ea2-4f13-a479-08ccb86193a4.png)
2. paste the above code into a python file

## Expected behavior
no error 

## Actual behavior

![image](https://user-images.githubusercontent.com/57028336/176589343-7b58999a-3cc4-4133-9c8b-d819832639f9.png)

**Comment by erictraut** [comment]
Yes, this is intentional. The only reason to use `assert_type` is if you're interested in type analysis. The same is true for `reveal_type`.

**Comment by DetachHead** [comment]
what if i'm not using pyright? i'm using mpy to check these type assertions

**Comment by erictraut** [comment]
@DetachHead, I noticed that whenever you file a bug, you "like" it, and then @KotlinIlsand also "likes" it within a second or two. I presume that you are one person using two different github accounts? Why do you do this? Just curious.

**Comment by erictraut** [comment]
> what if i'm not using pyright?

If you're using pylance, then you're also using pyright. Pylance is built on pyright.

**Comment by ethan-xd** [comment]
“+1 I also experiencing same issue ….”

**Comment by DetachHead** [comment]
> If you're using pylance, then you're also using pyright. Pylance is built on pyright.

i know that. but i'm not using pyright for type checking even though i'm still using its language server. i assume this is a supported use case, otherwise the `python.analysis.typeCheckingMode` setting wouldn't have an "off" option

**Comment by erictraut** [comment]
The "off" mode doesn't disable type analysis; it simply specifies defaults most of the user-configurable diagnostic rules to "none", suppressing their output. This mode is intended for people who are not interested in typing. If you're using `typing.type_assert()` in your code, then you are clearly interested in typing.

That said, I think it's reasonable to put `type_assert` under the "reportGeneralTypeIssue" diagnostic check. I've checked in this change, so in the future you won't see these errors if you set "typeCheckingMode" to "off".

**Comment by heejaechang** [comment]
This issue has been fixed in version 2022.7.21, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#2022721-6-July-2022)
